### ISSUE TITLE
Worked great from 1994 thru 1999 Delco Radios with the TMS370 microcontroller PCBs

### ISSUE THREAD

**Issue by drshock** [svschmitt/EC-PCB#1]
Thanks for this work, it inspired me to debug how the Theftlock codes are stored and generated in the Texas Instrument TMS370 microcontroller based radios and how the scan tool handshakes are implemented.   I did a YouTube video on how I utilized your shield and sample code to help analyze communications between the GM Tech2 dealer scan tool and Delco Radios here [Unlock 1990s GM Radio TheftLock Codes with Arduino](https://youtu.be/OJg6czxvJ2I)

I've included a link to this repo in the vids' description, as well as a shout out to your work here and elsewhere.

I did find that while the E&C bus continued thru 2002 in radios with the same cosmetic appearance, the TMS370 microcontroller was replaced in 2000.  And from that point on the protocol signal timings were changed.   So while I could find 100% repeatable success unlocking 1994 thru 1999 Delco radios, the 2000 thru 2002 ones were unreliable for this same technique as the bus protocols had changed.
 
It's been some time since I posted the video, but I thought about commenting here in case someone else using this work ran into this same incompatibility with the 2000 thru 2002 versions from Delco Electronics.

**Comment by person4268** [comment]
Did you ever figure out what the difference in signal timings was? I was about to go and start embarking on the idea of spoofing the OnStar module E&C communications to cleanly do aux-in without interfering with the CD changer. Also do really appreciate the video, it's the cleanest way I've seen to do it now that the phone line's gone.

**Comment by drshock** [comment]
No, I didn't invest any further time on this problem as I was only focused on the TMS370 centric circuit board type radios.
